atarashi/agents/semanticSearch.py

```diff
@@ -0,0 +1,208 @@
+        # Separate the comments into single line comments
+        file_comments = file_comments.split('\n')
+
+        # Separate each license text line by line and append them to one big list (used for matching)
+        # The license_index_map maps each line to the correct license index in the licenseList dataframe
+        license_index_map = {}
+        all_license_texts = []
+        for license_index, license_text in enumerate(self.licenseList['text']):
+            for line in license_text.split('\n'):
+                all_license_texts.append(line)
+                license_index_map[len(all_license_texts) - 1] = license_index
+        
+        # Perform first level fuzzy matching for all lines in the comments
+        results = []
+        fuzzy_similarity_matrix = np.zeros((len(file_comments), len(all_license_texts)))
+        for index, comment in enumerate(file_comments):
+            for i in range(len(all_license_texts)):
+                fuzzy_similarity_matrix[index][i] = fuzz.ratio(comment, all_license_texts[i]) 
+            max_score_index = np.argmax(fuzzy_similarity_matrix[index]) 
+            results.append(
+                        (
+                            fuzzy_similarity_matrix[index][max_score_index], 
+                            comment
+                        )
+                    ) 
+
+        # Try to append lines that match with a similarity score of more than 40% with one of the lines in 
+        # Any of the licenses. The goal is to get a bigger and bigger text chunk that matches to a bigger 
+        # text chunk in one of the license texts - The bigger the chunk, the more likely that this match is 
+        # correct.
+        appended_comments = []
+        appended_comment = []
+        for result in results:
+            if result[0] >= 40:
+                if appended_comment == [] and result[1] == '':
+                    continue
+                appended_comment.append(result[1])
+            else:
+                appended_comments.append(appended_comment)
+                appended_comment = []
+        
+        if len(appended_comment) > 0 and appended_comment not in appended_comments:
+            appended_comments.append(appended_comment)
+
+        # Attempt the final - second level license match with all the bigger text chunks
+        # In some licenses, the license header is used instead of the license text
+        # In case that a license has a license header, we match all chunks with both and take 
+        # the match with the highest similarity score
+        results = []
+        for appended_comment in appended_comments:
+            appended_comment = "\n".join(appended_comment)
+            fuzzy_similarity_matrix_2 = np.zeros(len(self.licenseList))
+            for i in range(len(self.licenseList)):
+                fuzzy_similarity_matrix_2[i] = fuzz.ratio(appended_comment, self.licenseList.loc[i, 'text'])
+                if pd.notna(licenseList.loc[i, 'license_header']):
```

Comment:
```suggestion
                if pd.notna(self.licenseList.loc[i, 'license_header']):
```

`licenseList` variable is not accessible